### PR TITLE
Run Test Framework against PSL & other updates

### LINKED ISSUE #65: Run Test Framework against PSL with new rewards
- [ ] Run framework
- [ ] Update sheet
- [ ] Add reporting against focused players list for Predictive Simulator Tab

### LINKED ISSUE #67: Run tournament simulator against PSL 2023
- [ ] Run tournament simulator for PSL 2023
- [ ] Tournament Simulato to include reporting for focus players
- [ ] Sp will need to provide the tournament schedule and playing XIs
- [ ] Tournament uses same format as ILT20

### PR DESCRIPTION
**Pull request by swati0308**
- Create a location to load up focus players, and add reporting on them 
- Add the ability to simulate PSL 2023
- Some tweaks to the static probability of players getting out, bowlers bowling an extra & the split of wides & no-balls
- Switched the bowler selection to a random process and cleaned up the complexity there 
- Added extra visualisations for predictive simulator & tournament simulator UIs

### CODE REVIEW

File: .FDm/config.local.toml [hunk 1]
```diff
@@ -12,6 +12,7 @@ player_file_name = "players.csv"
 repo_path="resources/prod/app_config/rewards_configuration"
 generated_path="data/generated/prod/rewards"
 file_name = "rewards.csv"
+focus_players_file = "focus_players.csv"
```
Comment (swati0308) [inline .FDm/config.local.toml:15]: File name for uploading a list of focus players for the app - for now, this is one static list of player keys.

File: .FDm/config.local.toml [hunk 2]
```diff
@@ -21,7 +22,7 @@ number_of_scenarios=4
 
 [player_outcome_predictor.tournament_simulator]
 number_of_scenarios=4
-data_path="resources/prod/app_config/tournament_simulator"
+data_path="resources/prod/app_config/tournament_simulator/psl20/"
```
Comment (swati0308) [inline .FDm/config.local.toml:25]: Set up the tournament simulator to simulate PSL

File: simulators/utils/predictive_utils.py [hunk 3]
```diff
@@ -94,7 +94,9 @@ def __init__(self,
         # TODO: These distributions will no longer be required once the inferential model comes into play.
 
         # Probability of a legal delivery
-        self.legal_delivery_distribution = sps.bernoulli(p=0.9)
```
Comment (swati0308) [inline simulators/utils/predictive_utils.py:97]: This section includes tweaks to the predictive simulator which may impact all models potentially

File: simulators/predictive_simulator.py [hunk 4]
```diff
@@ -102,7 +103,7 @@ def generate_innings(self, use_inferential_model):
 
         logging.debug("Initialising match state")
         match_state_dict = {}
-        self.simulated_matches_df.apply(lambda x: self.initialise_match_state(x, match_state_dict, playing_xi_df),
+        self.simulated_matches_df.swifter.apply(lambda x: self.initialise_match_state(x, match_state_dict, playing_xi_df),
```
Comment (swati0308) [inline simulators/predictive_simulator.py:106]: while it doesn't necessarily speed up the apply, it at least provides a helpful stat as it chugs along

File: simulators/utils/predictive_utils.py [hunk 5]
```diff
@@ -5,78 +5,7 @@
 import logging
 
 from inferential_models.batter_runs_models import BatterRunsModel
-
-
-def calculate_bowling_probabilities(all_innings_df) -> pd.DataFrame:
```
Comment (swati0308) [inline simulators/utils/predictive_utils.py:10]: Take away all the complex logic to calculate the next bowler, and resort to a simple random selection

**Review by gireesh-fractal — APPROVED** [review]

**Pull request merged**